python/pyiceberg/table/metadata.py

```diff
@@ -347,10 +345,10 @@ class TableMetadata:
 
     @staticmethod
     def parse_obj(data: dict) -> Union[TableMetadataV1, TableMetadataV2]:
-        if "format-version" not in data:
+        if "format-version" not in data and "format_version" not in data:
```

Comment:
This seems suspicious to me. When would this be called with `format_version` and not `format-version`?

Reply:
Yea, this is supspicious. I'm going to remove all the `pre_validators` because you can enter there the normal field name (`format_version`) and the alias (`format-version`) which makes it very confusing, clutters up the code, and makes it harder to comprehend the code.

Reply:
Okay, this is actually the only instance that we need this. Because we don't know if the aliases are already applied or not.

Reply:
Why don't we know if the aliases have already been applied?

Reply:
This is before anything pydantic, but the data might come from actual `json`, or from a `dict` that already has been parsed. In that case, the `-` will be replaced by `_`, and pydantic will handle this correctly. That said, I don't like it very much, and let me revert this.